bson/bsoncodec/default_value_decoders.go

```diff
@@ -98,7 +98,7 @@ func (dvd DefaultValueDecoders) RegisterDefaultDecoders(rb *RegistryBuilder) {
 		RegisterTypeMapEntry(bsontype.Decimal128, tDecimal).
 		RegisterTypeMapEntry(bsontype.MinKey, tMinKey).
 		RegisterTypeMapEntry(bsontype.MaxKey, tMaxKey).
-		RegisterTypeMapEntry(bsontype.Type(0), tD).
+		RegisterTypeMapEntry(bsontype.EmbeddedDocument, tD).
```

Comment:
I always found this a bit weird. That being said... I think we still need to care about `bsontype.Type(0)`. Existing code may pass that around, as wrong as it is.

Do we have any idea of what type of code would utilize Type(0)?

Reply:
Added back the `RegisterTypeMapEntry` call for `bsontype.Type(0)` both here and in the mgo registry.